projects/components/src/data-exporter/data-exporter.component.ts

```diff
@@ -163,28 +203,44 @@ export class DataExporterComponent implements OnInit {
 
     formGroup: FormGroup;
 
+    exportStage: LazyString;
+
+    optionsFormGroup = new FormGroup({
+        selectAll: new FormControl(true),
+        friendlyNames: new FormControl(true),
+        sanitize: new FormControl(false),
+    });
+
     onClickExport(): void {
+        this.exportStage = this.downloadingMessage;
         this._isRequestPending = true;
         this.dataExportRequest.emit({
             exportData: this.exportData.bind(this),
             updateProgress: this.updateProgress.bind(this),
-            selectedColumns: this.columns.filter(col => this.formGroup.controls[col.fieldName].value),
+            selectedColumns: this.selectedColumns,
         });
     }
 
-    onClickCheckAll(): void {
-        for (const column of this.columns) {
-            this.formGroup.controls[column.fieldName].setValue(true);
+    onClickCheckAll(value: boolean): void {
+        if (!this.optionsFormGroup.get('selectAll').value) {
```

Comment:
We also try to avoid using hard code strings to refer to controls and this seems to be the first time we need to hard code access to controls, yes we may not need it but doing the following would help other pieces of the code also.

```ts
// Create consts for the field names, they will only be used for initialization and in the control getter
const UiFields = createStringEnum([
    "field1",
    "field2",
    ...
]);

// Then we can use UiFields.field1  with some type safety (edited) 
// If bringing in createStringEnum is too much, we can just use
enum Fields = {
  field1: 'field1',
  field2: 'field2'
}
```
## HTML initialization
If you change the three places where you are returning the `formcontrol.value` such as [get isSelectAllEnabled()](https://github.com/vmware/vmware-cloud-director-ui-components/pull/95/files?file-filters%5B%5D=.json&file-filters%5B%5D=.properties&file-filters%5B%5D=.ts#diff-8ec3d6050a1425538628003d98cfe85dR234) to return the form control, they can also be used  to initialize the `[formControl]` in the HTML instead of using `[formControlName]`

## Testing considerations
You can now easily using the form control without having to write a wrapper for the checkbox since it's part of the component's API.

`comp.showFriendlyNamesControl.setValue(true)`
`expect(comp.showFriendlyNamesControl.value).toBe(false)`

Reply:
Changed to used the enum and getters to manage the form controls. 

And as far as the method onClickCheckAll goes, I gave the input type wrong in the code. The actual input type is a MouseEvent, which didn't seem to directly contain the information I needed. I removed the parameter to this method.